File: Knjigoteka.UI/knjigoteka_mobile/lib/models/borrowing_response.dart
class BorrowingResponse {
  final int id;
  final int bookId;
  final String title;
  final String? author;
  final DateTime borrowedAt;
  final DateTime dueDate;
  final DateTime? returnedAt;
  final String? photoEndpoint;

  BorrowingResponse({
    required this.id,
    required this.bookId,
    required this.title,
    this.author,
    required this.borrowedAt,
    required this.dueDate,
    this.returnedAt,
    this.photoEndpoint,
  });

  factory BorrowingResponse.fromJson(Map<String, dynamic> json) {
    return BorrowingResponse(
      id: json['id'],
      bookId: json['bookId'],
      title: json['bookTitle'] ?? '',
      author: json['author'], // može biti null
      borrowedAt: DateTime.parse(json['borrowedAt']),
      dueDate: DateTime.parse(json['dueDate']),
      returnedAt: json['returnedAt'] != null && json['returnedAt'] != ""
          ? DateTime.parse(json['returnedAt'])
          : null,
      photoEndpoint: json['photoEndpoint'],
    );
  }
}

File: Knjigoteka.Model/Responses/BorrowingResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Knjigoteka.Model.Responses
{
    public class BorrowingResponse
    {
        public int Id { get; set; }
        public int BookId { get; set; }
        public string BookTitle { get; set; } = null!;
        public string? Author { get; set; }
        public string? PhotoEndpoint { get; set; }
        public int UserId { get; set; }
        public string UserFullName { get; set; } = null!;
        public int BranchId { get; set; }
        public string BranchName { get; set; } = null!;
        public int? ReservationId { get; set; }
        public DateTime BorrowedAt { get; set; }
        public DateTime DueDate { get; set; }
        public DateTime? ReturnedAt { get; set; }
        public bool IsLate { get; set; }
    }

}
